sycl/source/context.cpp

```diff
@@ -37,13 +37,26 @@ context::context(const vector_class<device> &DeviceList,
   if (DeviceList.empty()) {
     throw invalid_parameter_error("DeviceList is empty.");
   }
-  if (DeviceList[0].is_host())
-    impl = std::make_shared<detail::context_impl>(DeviceList[0], AsyncHandler);
-  else
-    // TODO also check that devices belongs to the same platform
-    impl = std::make_shared<detail::context_impl>(DeviceList, AsyncHandler);
+  auto number = std::find_if_not(
```

Comment:
Why it is called `number`? If I understand correctly, it will contain some iterator, so, I suggest something like `NonHostDeviceIter` or just `NonHostDevice`